File: py/maxNestingDepthParentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        m = 0
        count = 0

        for c in s:
            if c == "(": count +=1
            elif c == ")": count -=1
            else: pass
            m = max(count, m)
        
        return m

File: cpp/maxNestingDepthParentheses.cpp
#include <string>

using namespace std;

class Solution {
public:
    int maxDepth(string s) {
        int maxc = 0;
        int count = 0;
        for (char c : s) {
            if (c == '(') count++;
            else if (c == ')') count --;
            else {}
            maxc = max(count, maxc);
        }
        return maxc;
    }
};